### PR TITLE
Feature/wish list filters

### PR DESCRIPTION
**Pull request by CowboyLynk**
#4

### CODE REVIEW

File: courses_database/static/js/init.js [hunk 1]
```diff
@@ -37,4 +43,130 @@ function tablesInit(){
...
+
+    let min_credits = document.getElementById('min_credits');
+    let max_credits = document.getElementById('max_credits');
+
+    min_credits.addEventListener('change', function() {
+        credit_slider.noUiSlider.set([this.value, null]);
+    }, {passive: true});
+    max_credits.addEventListener('change', function() {
+        credit_slider.noUiSlider.set([null, this.value]);
+    }, {passive: true});
```
Comment (DanielMcCrystal) [inline courses_database/static/js/init.js]: What does "passive" mean here?

File: courses_database/static/js/main.js [hunk 2]
```diff
@@ -16,7 +22,12 @@ function updateSchedules(is_async) {
     if (typeof(is_async)==='undefined') is_async = true;  // default value for is_async
 
     // TODO: when schedule filters are implemented, update this line to get those filters
-    let filters = {};
+    let filters = $('#wish_list_filters').serializeArray().reduce(function(obj, item) {
+        obj[item.name] = item.value;
+        return obj;
+    }, {});
```
Comment (DanielMcCrystal) [inline courses_database/static/js/main.js]: Can you explain what's going on here?

File: courses_database/static/js/init.js (hunk 1 above)
Comment (CowboyLynk) [reply courses_database/static/js/init.js]: I was trying to do this to get rid of some warnings in Chrome: https://developers.google.com/web/updates/2016/06/passive-event-listeners

But it didn't work.

File: courses_database/static/js/main.js (hunk 2 above)
Comment (CowboyLynk) [reply courses_database/static/js/main.js]: the syntax is super confusing but basically, it gets all the filters from the filter form. 'reduce' takes in a function where the first argument always gets passed into the next call. So for example, 'obj' is always the same object whereas 'item' is just iterating through each item in the array.

Comment (DanielMcCrystal) [reply courses_database/static/js/main.js]: Ah I think I get it. So it's basically just converting the HTML form into a JS dictionary.

Comment (CowboyLynk) [reply courses_database/static/js/main.js]: Yeah exactly

**Pull request merged**